django/db/backends/oracle/creation.py

```diff
@@ -40,6 +40,11 @@ def _create_test_db(self, verbosity=1, autoclobber=False, keepdb=False):
                 if keepdb:
                     return
                 sys.stderr.write("Got an error creating the test database: %s\n" % e)
+                if 'ORA-01031' in str(e):
+                    sys.stderr.write(
+                        "It appears the main user has insufficient privileges "
+                        "to create test database.\n")
+                    sys.exit(2)
```

Comment:
I think you're probably right about these four lines. The only reason I see for a separate PR, though, is if that PR were to include a battery of tests (using mocking as @timgraham suggested) for the different combinations of pre-existence of user & tablespace and values of the `keepdb` flag.